datafusion/functions-aggregate/src/variance.rs

```diff
@@ -470,7 +470,7 @@ impl VarianceGroupsAccumulator {
 
         if let StatsType::Sample = self.stats_type {
             counts.iter_mut().for_each(|count| {
-                *count -= 1;
+                *count = count.saturating_sub(1);
```

Comment:
Should we return error if count is already 0?

Reply:
No, I think producing null is the expected behavior. Note that we use the counts to calculate the null array after this.